lua/harpoon/ui.lua

```diff
@@ -154,6 +153,7 @@ function HarpoonUI:save()
     local list = Buffer.get_contents(self.bufnr)
     Logger:log("ui#save", list)
     self.active_list:resolve_displayed(list)
+    require("harpoon"):sync()
```

Comment:
Actually sync when saving, so that the only save isn't run on `VimLeavePre` (which seems to be inconsistent)

Reply:
Nevermind, this is actually necessary, but I moved it to `resolve_displayed` since that's where the data is actually modified.